File: bakerasr/src/main/java/com/baker/speech/asr/bean/BakerException.java
package com.baker.speech.asr.bean;

import androidx.annotation.Nullable;

import java.io.IOException;

/**
 * @author hsj55
 * 2021/2/2
 */
public class BakerException extends IOException {
    private String code;
    private String message;

    public BakerException(String code, String message) {
        this.code = code;
        this.message = message;
    }

    public BakerException(String code) {
        this.code = code;
    }

    public BakerException(Exception e) {
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    @Nullable
    @Override
    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    @Override
    public String toString() {
        return "BakerException{" +
                "code=" + code +
                ", message='" + message + '\'' +
                '}';
    }
}

File: bakervpr/src/main/java/com/baker/sdk/vpr/bean/BakerException.kt
package com.baker.sdk.vpr.bean

import java.io.IOException
import java.lang.Exception

/**
 * @author hsj55
 * 2021/2/2
 */
class BakerException : IOException {
    var code: String? = null
    override var message: String? = null

    constructor(code: String?, message: String?) {
        this.code = code
        this.message = message
    }

    constructor(code: String?) {
        this.code = code
    }

    constructor(e: Exception?)

    override fun toString(): String {
        return "BakerException{" +
                "code=" + code +
                ", message='" + message + '\'' +
                '}'
    }
}